src/compiler/Stage/Desugar.elm

```diff
@@ -262,11 +234,50 @@ qualifiedVarInImportedModule modules maybeModuleName varName =
         |> Maybe.withDefault Nothing
 
 
-qualifiedVarInAliasedModule : Modules Frontend.Expr -> Module Frontend.Expr -> Maybe ModuleName -> VarName -> Maybe ModuleName
+qualifiedVarInAliasedModule : Modules Frontend.LocatedExpr -> Module Frontend.LocatedExpr -> Maybe ModuleName -> VarName -> Maybe ModuleName
 qualifiedVarInAliasedModule modules thisModule maybeModuleName varName =
     let
         unaliasedModuleName =
             Maybe.andThen (Common.unalias thisModule) maybeModuleName
     in
     -- Reusing the existing functionality. TODO is this a good idea?
     qualifiedVarInImportedModule modules unaliasedModuleName varName
+
+
+locatedResultReturn : Frontend.LocatedExpr -> Canonical.Expr -> Result DesugarError Canonical.LocatedExpr
+locatedResultReturn located expr =
+    Ok
+        (Located.replaceWith
+            expr
+            located
+        )
+
+
+locatedResultMap : Frontend.LocatedExpr -> (a -> Canonical.Expr) -> Result DesugarError a -> Result DesugarError Canonical.LocatedExpr
+locatedResultMap located fn =
+    Result.map
+        (\expr ->
+            Located.replaceWith
+                (fn expr)
+                located
+        )
+
+
+locatedResultMap2 : Frontend.LocatedExpr -> (a -> b -> Canonical.Expr) -> Result DesugarError a -> Result DesugarError b -> Result DesugarError Canonical.LocatedExpr
+locatedResultMap2 located fn =
+    Result.map2
+        (\expr1 expr2 ->
+            Located.replaceWith
+                (fn expr1 expr2)
+                located
+        )
+
+
+locatedResultMap3 : Frontend.LocatedExpr -> (a -> b -> c -> Canonical.Expr) -> Result DesugarError a -> Result DesugarError b -> Result DesugarError c -> Result DesugarError Canonical.LocatedExpr
+locatedResultMap3 located fn =
+    Result.map3
+        (\expr1 expr2 expr3 ->
+            Located.replaceWith
+                (fn expr1 expr2 expr3)
+                located
+        )
```

Comment:
I'm just flagging these functions to later experiment with whether it wouldn't be nicer to inline them in the `let ...` they're used in. That doesn't need to happen in this PR though - I can do that on my own later.